pkg/binding/binding.go

```diff
@@ -146,3 +157,178 @@ func (o *BindingClient) GetServiceInstances() (map[string]unstructured.Unstructu
+		if err != nil {
+			return nil, err
+		}
+
+		u := unstructured.Unstructured{}
+		if err := yaml.Unmarshal([]byte(strCRD), &u.Object); err != nil {
+			return nil, err
+		}
+
+		switch u.GetObjectKind().GroupVersionKind() {
+		case bindingApi.GroupVersionKind:
+
+			var sbo bindingApi.ServiceBinding
+			err := o.kubernetesClient.ConvertUnstructuredToResource(u, &sbo)
+			if err != nil {
+				return nil, err
+			}
+
+			sb := api.ServiceBindingFromBinding(sbo)
+			sb.Status, err = o.getStatusFromBinding(sb.Name)
+			if err != nil {
+				return nil, err
+			}
+
+			result = append(result, sb)
+
+		case specApi.GroupVersion.WithKind("ServiceBinding"):
+
+			var sbc specApi.ServiceBinding
+			err := o.kubernetesClient.ConvertUnstructuredToResource(u, &sbc)
+			if err != nil {
+				return nil, err
+			}
+
+			sb := api.ServiceBindingFromSpec(sbc)
+			sb.Status, err = o.getStatusFromSpec(sb.Name)
+			if err != nil {
+				return nil, err
+			}
+
+			result = append(result, sb)
+
+		}
+	}
+	return result, nil
+}
+
+// GetBinding returns the ServiceBinding retource with the given name
+// from the cluster, from group binding.operators.coreos.com/v1alpha1 or servicebinding.io/v1alpha3
+func (o *BindingClient) GetBinding(name string) (api.ServiceBinding, error) {
+
+	bindingSB, err := o.kubernetesClient.GetBindingServiceBinding(name)
+	if err == nil {
+		sb := api.ServiceBindingFromBinding(bindingSB)
+		sb.Status, err = o.getStatusFromBinding(bindingSB.Name)
+		if err != nil {
+			return api.ServiceBinding{}, err
+		}
+		return sb, nil
+	}
+	if err != nil && !kerrors.IsNotFound(err) {
+		return api.ServiceBinding{}, err
+	}
+
+	specSB, err := o.kubernetesClient.GetSpecServiceBinding(name)
+	if err == nil {
+		sb := api.ServiceBindingFromSpec(specSB)
+		sb.Status, err = o.getStatusFromSpec(specSB.Name)
+		if err != nil {
+			return api.ServiceBinding{}, err
+		}
+		return sb, nil
+	}
+
+	// In case of notFound error, this time we return the error
+	if kerrors.IsNotFound(err) {
+		return api.ServiceBinding{}, fmt.Errorf("ServiceBinding %q not found", name)
+	}
+	return api.ServiceBinding{}, err
+}
+
+// getStatusFromBinding returns status information from a ServiceBinding in the cluster
+// from group binding.operators.coreos.com/v1alpha1
+func (o *BindingClient) getStatusFromBinding(name string) (*api.ServiceBindingStatus, error) {
+	bindingSB, err := o.kubernetesClient.GetBindingServiceBinding(name)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	if injected := meta.IsStatusConditionTrue(bindingSB.Status.Conditions, bindingApis.InjectionReady); !injected {
+		return nil, nil
+	}
+
+	secretName := bindingSB.Status.Secret
+	secret, err := o.kubernetesClient.GetSecret(secretName, o.kubernetesClient.GetCurrentNamespace())
+	if err != nil {
+		return nil, err
+	}
+
+	if bindingSB.Spec.BindAsFiles {
+		bindings := make([]string, 0, len(secret.Data))
```

Comment:
This declaration could be made outside the `if` block so that it doesn't need to be done twice.